tests/unit/api/c/receiver/test_receiver.cpp

```diff
@@ -66,6 +67,109 @@ class TestReceiver : public ::testing::Test
+
+    clear_term_set_list_fake.custom_fake = [](TerminationSet* list) { EXPECT_EQ(list, nullptr); };
+
+    switch (ip_supported)
+    {
+      case kSacnIpV4Only:
+        sacn_remove_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_socket_t* socket, bool) {
+          ASSERT_NE(socket, nullptr);
+          EXPECT_NE(*socket, CHANGE_UNIVERSE_WORKS_FIRST_IPV6_SOCKET);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+        };
+        sacn_add_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_iptype_t ip_type,
+                                                       uint16_t universe, const EtcPalMcastNetintId*, size_t,
+                                                       etcpal_socket_t* socket) {
+          EXPECT_NE(ip_type, kEtcPalIpTypeV6);
+          EXPECT_NE(ip_type, kEtcPalIpTypeInvalid);
+          EXPECT_EQ(universe, CHANGE_UNIVERSE_WORKS_SECOND_UNIVERSE);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+          EXPECT_NE(socket, nullptr);
+          return kEtcPalErrOk;
+        };
+        break;
+      case kSacnIpV6Only:
+        sacn_remove_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_socket_t* socket, bool) {
+          ASSERT_NE(socket, nullptr);
+          EXPECT_NE(*socket, CHANGE_UNIVERSE_WORKS_FIRST_IPV4_SOCKET);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+        };
+        sacn_add_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_iptype_t ip_type,
+                                                       uint16_t universe, const EtcPalMcastNetintId*, size_t,
+                                                       etcpal_socket_t* socket) {
+          EXPECT_NE(ip_type, kEtcPalIpTypeV4);
+          EXPECT_NE(ip_type, kEtcPalIpTypeInvalid);
+          EXPECT_EQ(universe, CHANGE_UNIVERSE_WORKS_SECOND_UNIVERSE);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+          EXPECT_NE(socket, nullptr);
+          return kEtcPalErrOk;
+        };
+        break;
+      case kSacnIpV4AndIpV6:
+        sacn_remove_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_socket_t* socket, bool) {
+          ASSERT_NE(socket, nullptr);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+        };
+        sacn_add_receiver_socket_fake.custom_fake = [](sacn_thread_id_t thread_id, etcpal_iptype_t ip_type,
+                                                       uint16_t universe, const EtcPalMcastNetintId*, size_t,
+                                                       etcpal_socket_t* socket) {
+          EXPECT_NE(ip_type, kEtcPalIpTypeInvalid);
+          EXPECT_EQ(universe, CHANGE_UNIVERSE_WORKS_SECOND_UNIVERSE);
+          EXPECT_NE(get_recv_thread_context(thread_id), nullptr);
+          EXPECT_NE(socket, nullptr);
+          return kEtcPalErrOk;
+        };
+    }
+
+    etcpal_error_t change_universe_result =
```

Comment:
If you do follow Sam's suggestion, lines 147 on could be refactored into a function that your individual test functions can call (e.g. PerformUniverseChangeTest)